pkg/pmem-csi-operator/controller/deployment/controller_driver.go

```diff
@@ -1026,25 +910,14 @@ func (d *pmemCSIDeployment) getControllerStatefulSet(ss *appsv1.StatefulSet) {
 				},
 				Volumes: []corev1.Volume{
 					{
-						Name: "plugin-socket-dir",
-						VolumeSource: corev1.VolumeSource{
-							EmptyDir: &corev1.EmptyDirVolumeSource{},
-						},
-					},
-					{
-						Name: "registry-cert",
+						Name: "webhook-cert",
 						VolumeSource: corev1.VolumeSource{
 							Secret: &corev1.SecretVolumeSource{
+								// TODO: replace name, create secret via cert-manager
```

Comment:
This is a bit more complicated. When deploying via OLM, we need OLM to create the certificates. My understanding is that this can be done for "normal" webhooks, just not for conversion webhooks.

For deployment of the operator via YAML files we need to document how to set up those certs.

In both cases, re-configuring the cluster to use the webhooks is something that needs to be done manually.

If the user of the operator doesn't need the scheduler extensions, then it's still okay to start the statefulset unconditionally. The pod simply won't run.

Reply:
> This is a bit more complicated. When deploying via OLM, we need OLM to create the certificates. My understanding is that this can be done for "normal" webhooks, just not for conversion webhooks.

I doubt if OLM can creates certificates for services created by the operator for the CRs, though it provides service certificates for operator itself.

Reply:
@pohly How this scheduler extension and the Pod mutation works in the context of multiple PmemCSIDeployments? I think we should /cannot support multiple deployments.

Reply:
For the mutating pod webhook: it must use a resource name which includes the driver name. I thought it already did this, but currently it doesn't (https://github.com/intel/pmem-csi/blob/692818be4679e826fc7a98e77ad0eeb61b40dba5/pkg/scheduler/mutate_pod.go#L31).

For the scheduler extender: it must be triggered by the same resource name (driver name to be added) and then check only pods that belong to that instance (already implemented: https://github.com/intel/pmem-csi/blob/82bcde130b32d006b0da5b11729373a91ae575d1/pkg/scheduler/capacity.go#L103-L106)

Reply:
> I doubt if OLM can creates certificates for services created by the operator for the CRs, though it provides service certificates for operator itself.

When we discussed it this morning, my conclusion was that the user needs to supply TLS credentials and that the webhooks need to be disabled if not provided.

Reply:
> In both cases, re-configuring the cluster to use the webhooks is something that needs to be done manually.

Have I mentioned how much I hate configuring a scheduler extension? :cry: 

The way we currently do it is by creating the certificates first, then setting up the "kubeadm init" configuration so that those certificates can be used by `kube-scheduler`. This no longer works when creating those certificates *inside* the cluster with a tool like cert-manager.

It is possible to set up the cluster without the extension enabled, then create certificates, and then reconfigure `kube-scheduler`. `kubeadm upgrade` supports that. But this looks complicated. I think I'll stick with the current approach.